include/nlohmann/detail/macro_scope.hpp

```diff
@@ -133,6 +133,14 @@
     #endif
 #endif
 
+#ifndef JSON_HAS_STATIC_RTTI
+    #if defined(_HAS_STATIC_RTTI) && _HAS_STATIC_RTTI != 0
+        #define JSON_HAS_STATIC_RTTI 1
+    #else
+        #define JSON_HAS_STATIC_RTTI 0
```

Comment:
It looks like `_HAS_STATIC_RTTI` is a MS extension and when it's set to `0`, certain functionality is disabled.  Shouldn't this then default to 1:

    #if !defined(_HAS_STATIC_RTTI) || _HAS_STATIC_RTTI != 0
        #define JSON_HAS_STATIC_RTTI 1
    #else
        #define JSON_HAS_STATIC_RTTI 0
    #endif